faunadb/query.py

```diff
@@ -40,6 +43,43 @@ def quote(expr):
   """See the `docs <https://faunadb.com/documentation/queries#basic_forms>`__."""
   return {"quote": expr}
 
+thread_local = local()
+thread_local.lambda_auto_var_number = 0
+
+def lambda_auto(lambda_body):
+  """
+  See the `docs <https://faunadb.com/documentation/queries#basic_forms>`__.
+  This form generates the names of lambda parameters for you, and is called like::
+
+      query.lambda_auto(lambda a: query.add(a, a))
+      # Produces: {
+      #  "lambda": "auto0",
+      #  "expr": {"add": ({"var": "auto0"}, {"var": "auto0"})}
+      # }
+
+  Query functions requiring lambdas can be passed raw lambdas
+  without explicitly calling ``lambda_auto``.
+  For example: ``query.map(lambda a: query.add(a, 1), collection)``.
+
+  You can also use :any:`lambda_expr` directly.
+
+  :param lambda_body: Takes a :any:`var` and uses it to construct an expression.
+  """
+  var_name = "auto%i" % thread_local.lambda_auto_var_number
+  thread_local.lambda_auto_var_number += 1
+
+  # Make sure lambda_auto_var_number returns to its former value even if there are errors.
+  try:
+    return lambda_expr(var_name, lambda_body(var(var_name)))
+  finally:
+    thread_local.lambda_auto_var_number -= 1
+
+def to_lambda(value):
+  """ If ``value`` is a lambda, converts it to a query using :any:`lambda_auto`."""
+  if isinstance(value, FunctionType):
```

Comment:
I mean either have them explicitly use one or the other, not have a `to_lambda` catch all.